lib/class-controller.php

```diff
@@ -31,14 +31,105 @@ class Controller {
 	 */
 	private static Controller $instance;
 
+	/**
+	 * Enable aggregations based on WP Categories.
+	 *
+	 * @param array $args Arguments to pass to the adapter's facet configuration.
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_category_aggregation( array $args = [] ): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_category_aggregation();
+			$defaults = [
+				'count'     => 1000,
+				'name'      => 'Categories',
+				'startOpen' => false,
+				'type'      => 'category',
+			];
+
+			$args = wp_parse_args( $args, $defaults );
+			$this->adapter->add_facet_config( $args );
+		}
+		return $this;
+	}
+
+	/**
+	 * Enable faceting on empty search query strings.
+	 *
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_empty_search_faceting(): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_empty_search_faceting();
+		}
+		return $this;
+	}
+
+	/**
+	 * Enable aggregations based on WP Tags.
+	 *
+	 * @param array $args Arguments to pass to the adapter's facet configuration.
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_tag_aggregation( array $args = [] ): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_tag_aggregation();
+			$defaults = [
+				'count'     => 1000,
+				'name'      => 'Tags',
+				'startOpen' => false,
+				'taxonomy'  => 'post_tag',
```

Comment:
I wonder if it's possible to make this generic per taxonomy - we should be able to pull information about the taxonomy from its slug, so we could do something like `enable_taxonomy_aggregation( 'post_tag' )` which would get info about the `post_tag` taxonomy which would include its label (`Tags`), and do the same with categories and custom taxonomies, rather than having a special handler for categories and tags.